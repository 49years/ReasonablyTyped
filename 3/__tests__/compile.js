const fs = require("fs");
const path = require("path");
const babel = require("../babel");

const fixtures = path.join(__dirname, "fixtures");
const fixture = file => {
  return fs.readFileSync(path.join(fixtures, file));
};

const testFiles = fs.readdirSync(fixtures).map(file => {
  it(file, () => {
    const input = fixture(file);
    const code = babel(input);
    expect(code).toMatchSnapshot();
  });
});
