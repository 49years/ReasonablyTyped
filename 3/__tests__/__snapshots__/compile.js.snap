// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function.js 1`] = `"declare function add(a: number, b: number): number {}"`;

exports[`function-with-union.js 1`] = `
"declare type buttonSize = \\"small\\" | \\"default\\" | \\"large\\";

declare function getCorrectSize(size: \\"small\\" | \\"default\\" | \\"large\\"): number {}"
`;

exports[`interface.js 1`] = `
"declare interface Props {
  innerRef?: elementRef<*>,
  htmlType?: \\"submit\\" | \\"button\\",
  text: react$Node,
  type: buttonType,
  onClick?: (SyntheticEvent<*>) => void | Promise<*>,
  className?: string,
  loading?: boolean,
  disabled: boolean,
  size: buttonSize,
  style?: object,
}"
`;

exports[`interface-with-union.js 1`] = `
"declare type buttonSize = \\"small\\" | \\"default\\" | \\"large\\";
declare interface Props {
  size: \\"small\\" | \\"default\\" | \\"large\\"
}"
`;
